### PR TITLE
Customizable wandb data

### LINKED ISSUE #162: customizable wandb data
Right now, the wandb run only saves training loss. That's helpful, but it would be even more helpful if a user of the pipeline could save other information.

**Acceptance Criteria**:
- There is some way to specify extra information that gets saved in the wandb run

**Possible Design**:
- Do #90 
- Have an extra member on the model output that is something like "training metrics" that is a simple dict.
- Pass those training metrics into the on_train_batch_end callback and have the wandb callback include that

### PR DESCRIPTION
**Pull request by JansenNye**
Tested by running training run with the implemented GradNormMetrics example and confirming that wandb logs grad norm.
Closes #162

### CODE REVIEW

File: src/mirror/callbacks/wandb_callback.py [hunk 1]
```diff
@@ -56,12 +59,18 @@ def on_fit_start(
...
         loss: float,
         **kwargs,
     ):
         if self.run:
             self.step += 1
-            self.run.log({"train/loss": loss}, step=self.step)
+            extra_metrics = (
+                self.extra_metrics_getter.get_metrics(model)
+                if self.extra_metrics_getter
+                else {}
```
Comment (morrisoncheney) [inline src/mirror/callbacks/wandb_callback.py]: hmmm, interesting notation. Nice!

File: docs/mirror onboarding/mirror onboarding.md
Comment (morrisoncheney) [inline docs/mirror onboarding/mirror onboarding.md:1]: Nice work updating the documentation. It's clutch that you remembered to do that.

**Review by morrisoncheney — APPROVED** [review]
Looks good!

File: src/mirror/metrics/grad_norm_metrics.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+from mirror.models.mirror_model import MirrorModel
+
+
+class GradNormMetrics(ExtraMetricsGetter):
+    def get_metrics(self, model: MirrorModel) -> dict:
+        total_sq = 0.0
+        for p in model.parameters():
+            if p.grad is not None:
+                total_sq += p.grad.detach().pow(2).sum().item()
+        return {"grad_norm": total_sq ** 0.5}
```
Comment (Plyb) [inline src/mirror/metrics/grad_norm_metrics.py]: Has this been tested on multiple nodes? iirc model.parameters() behaves kind of weirdly when you're doing things like fsdp.

I also just want to check that we're not going to run into any numerical instability from summing up all of the squared parameters like this. One option would be to put it all in a tensor together and use pytorch's built in norm methods.

**Review by Plyb — CHANGES_REQUESTED** [review]
a couple questions

File: src/mirror/metrics/grad_norm_metrics.py (hunk 2 above)
Comment (JansenNye) [reply src/mirror/metrics/grad_norm_metrics.py]: Ah, yeah, I'll fix these

**Review by Plyb — APPROVED** [review]
